receiver/mongodbatlasreceiver/internal/metadata/metric_name_mapping.go

```diff
@@ -15,707 +15,856 @@
+			mb.RecordMongodbatlasDiskPartitionUtilizationMaxDataPoint(ts, float64(*dp.Value), AttributeDiskStatus.Free)
+		}
+	case "DISK_PARTITION_SPACE_PERCENT_USED":
+		return func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+			mb.RecordMongodbatlasDiskPartitionUtilizationAverageDataPoint(ts, float64(*dp.Value), AttributeDiskStatus.Used)
+		}
+	case "MAX_DISK_PARTITION_SPACE_PERCENT_USED":
+		return func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+			mb.RecordMongodbatlasDiskPartitionUtilizationMaxDataPoint(ts, float64(*dp.Value), AttributeDiskStatus.Used)
+		}
+
+	// Process Database Measurements (https://docs.atlas.mongodb.com/reference/api/process-disks-measurements/)
+	case "DATABASE_COLLECTION_COUNT":
+		return func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+			mb.RecordMongodbatlasDbCountsDataPoint(ts, float64(*dp.Value), AttributeObjectType.Collection)
+		}
+	case "DATABASE_INDEX_COUNT":
+		return func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+			mb.RecordMongodbatlasDbCountsDataPoint(ts, float64(*dp.Value), AttributeObjectType.Index)
+		}
+	case "DATABASE_EXTENT_COUNT":
+		return func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+			mb.RecordMongodbatlasDbCountsDataPoint(ts, float64(*dp.Value), AttributeObjectType.Extent)
 		}
+	case "DATABASE_OBJECT_COUNT":
+		return func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+			mb.RecordMongodbatlasDbCountsDataPoint(ts, float64(*dp.Value), AttributeObjectType.Object)
+		}
+	case "DATABASE_VIEW_COUNT":
+		return func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+			mb.RecordMongodbatlasDbCountsDataPoint(ts, float64(*dp.Value), AttributeObjectType.View)
+		}
+	case "DATABASE_AVERAGE_OBJECT_SIZE":
+		return func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+			mb.RecordMongodbatlasDbSizeDataPoint(ts, float64(*dp.Value), AttributeObjectType.Object)
+		}
+	case "DATABASE_STORAGE_SIZE":
+		return func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+			mb.RecordMongodbatlasDbSizeDataPoint(ts, float64(*dp.Value), AttributeObjectType.Storage)
+		}
+	case "DATABASE_INDEX_SIZE":
+		return func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+			mb.RecordMongodbatlasDbSizeDataPoint(ts, float64(*dp.Value), AttributeObjectType.Index)
+		}
+	case "DATABASE_DATA_SIZE":
+		return func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+			mb.RecordMongodbatlasDbSizeDataPoint(ts, float64(*dp.Value), AttributeObjectType.Data)
+		}
+
 	default:
-		return nil, fmt.Errorf("unrecognized data type for metric '%s'", meas.Name)
+		return nil
 	}
+}
 
-	return &m, nil
+func MeasurementsToMetric(mb *MetricsBuilder, meas *mongodbatlas.Measurements, buildUnrecognized bool) error {
+	recordFunc := getRecordFunc(meas.Name)
```

Comment:
My idea for the switch case was to avoid passing functions. Something like this:

```golang
for _, point := range meas.DataPoints {
    if point.Value != nil {
        curTime, err := time.Parse(time.RFC3339, point.Timestamp)
        if err != nil {
            return err
        }
        record(mb, meas, point, curTime)
    }
}
```

And inside `record` we use switch case to make particular call.

Not needed for this PR tho. Can be done separately.